### PR TITLE
[AskMode] Add support of `IsTrue` and `IsFalse` user definded operator to VB

### LINKED ISSUE #9127: Unexpected value 'IsTrue' of type 'Microsoft.CodeAnalysis.VisualBasic.UnaryOperatorKind'
Here's several tests to repro this crash (with OperationWalker hooked up)
```
Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.UserDefinedConversions.BooleanExpressions3
Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.UserDefinedConversions.BooleanExpressions5
Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.UserDefinedConversions.BooleanExpressions0
```
Call stack:
```
Test 'Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.UserDefinedConversions.BooleanExpressions3' failed: Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : Unexpected value 'IsTrue' of type 'Microsoft.CodeAnalysis.VisualBasic.UnaryOperatorKind'

	ThrowingTraceListener.cs(25,0): at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
	at System.Diagnostics.TraceListener.Fail(String message)
	at System.Diagnostics.TraceInternal.Fail(String message)
	at System.Diagnostics.Debug.Assert(Boolean condition, String message)
	InternalUtilities\ExceptionUtilities.cs(13,0): at Roslyn.Utilities.ExceptionUtilities.UnexpectedValue(Object o)
	BoundTree\Expression.vb(658,0): at Microsoft.CodeAnalysis.VisualBasic.BoundUserDefinedUnaryOperator.get_IUnaryOperatorExpression_UnaryOperationKind()
	c:\GitHub\roslyn-internal\Open\src\Test\Utilities\Shared\Compilation\TestOperationWalker.cs(351,0): at Microsoft.CodeAnalysis.Test.Utilities.TestOperationWalker.VisitUnaryOperatorExpression(IUnaryOperatorExpression operation)
	BoundTree\Expression.vb(668,0): at Microsoft.CodeAnalysis.VisualBasic.BoundUserDefinedUnaryOperator.Accept(OperationVisitor visitor)
	Compilation\OperationWalker.cs(34,0): at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
	c:\GitHub\roslyn-internal\Open\src\Test\Utilities\Shared\Compilation\TestOperationWalker.cs(41,0): at Microsoft.CodeAnalysis.Test.Utilities.TestOperationWalker.Visit(IOperation operation)
	Compilation\OperationWalker.cs(88,0): at Microsoft.CodeAnalysis.Semantics.OperationWalker.VisitIfStatement(IIfStatement operation)
	BoundTree\Statement.vb(75,0): at Microsoft.CodeAnalysis.VisualBasic.BoundIfStatement.Accept(OperationVisitor visitor)
	Compilation\OperationWalker.cs(34,0): at Microsoft.CodeAnalysis.Semantics.OperationWalker.Visit(IOperation operation)
	c:\GitHub\roslyn-internal\Open\src\Test\Utilities\Shared\Compilation\TestOperationWalker.cs(41,0): at Microsoft.CodeAnalysis.Test.Utilities.TestOperationWalker.Visit(IOperation operation)
	CompilationTestUtils.vb(279,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.CompilationUtils.WalkOperationTrees(VisualBasicCompilation compilation)
	CompilationTestUtils.vb(230,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.CompilationUtils.CreateCompilationWithReferences(IEnumerable`1 sourceTrees, IEnumerable`1 references, VisualBasicCompilationOptions options, String assemblyName)
	CompilationTestUtils.vb(200,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.CompilationUtils.CreateCompilationWithReferences(XElement sources, IEnumerable`1 references, VisualBasicCompilationOptions options, IEnumerable`1& spans, VisualBasicParseOptions parseOptions)
	BasicTestBase.vb(162,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, IEnumerable`1 allReferences, String expectedOutput, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, Boolean verify)
	BasicTestBase.vb(129,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, String expectedOutput, MetadataReference[] additionalRefs, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, Boolean verify, Boolean useLatestFramework)
	BasicTestBase.vb(51,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, XCData expectedOutput, MetadataReference[] additionalRefs, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, Boolean verify)
	Semantics\UserDefinedConversions.vb(3709,0): at Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.UserDefinedConversions.BooleanExpressions3()

```

### PR DESCRIPTION
**Pull request by genlu**
This fixes #9127. 
This will also avoid any other unexpected value in IOperation to crash the compiler.

@JohnHamby @dotnet/roslyn-interactive @AlekseyTs @MattGertz

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs [hunk 1]
```diff
@@ -1724,5 +1724,48 @@ struct S
...
+        if (x && y) { }
+        else if (x) { }
+    }
+}
+";
+            CreateCompilationWithMscorlib45(source)
+            .VerifyDiagnostics()
+            .VerifyAnalyzerDiagnostics(new DiagnosticAnalyzer[] { new TrueFalseUnaryOperationTestAnalyzer() }, null, null, false,
+                Diagnostic(TrueFalseUnaryOperationTestAnalyzer.UnaryTrueDescriptor.Id, "x && y").WithLocation(29, 13),
+                Diagnostic(TrueFalseUnaryOperationTestAnalyzer.UnaryTrueDescriptor.Id, "x").WithLocation(30, 18));
```
Comment (JohnHamby) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Neither this test nor the VB version tests getting a unary false operator.

**Comment by JohnHamby** [comment]
I sign off, with the slight caveat that the tests should cover the IsFalse case if possible.

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs (hunk 1 above)
Comment (genlu) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs]: Yup, I have created an issue for this #9309

**Comment by genlu** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/4565/
// Retest reason: This has nothing to do with this PR, probably a flaky test, I will rerun and test and create an issue if it passed.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by genlu** [comment]
@JohnHamby Just a heads up, this might cause a minor conflict with #9318.

**Comment by genlu** [comment]
retest this please.

**Pull request merged**